browser_use/agent/message_manager/utils.py

```diff
@@ -15,6 +15,29 @@
 
 logger = logging.getLogger(__name__)
 
+"""
+List of model names that do not support tool/function calling capabilities.
+Each entry can be either:
+- A string representing the exact model name
+- A list of strings where all parts must be present in the model name
+"""
+MODELS_WITHOUT_TOOL_SUPPORT = [
+	"deepseek-reasoner",
+	["deepseek-r1"],
+	["gemma", "-it"],
+]
```

Comment:
sorry to nit but I think this would be clearer/simpler as a list of regexes.

```suggestion
MODELS_WITHOUT_TOOL_SUPPORT = [
	r"deepseek-reasoner",
	r"deepseek-r1",
	r"gemma.*-it",
]
```

Reply:
You're right. This is a much better sollution. I've pushed a fix